platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/builders/TaskSchedulerConfiguration.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.wiring.schedulers.builders;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * Configures a task scheduler.
+ *
+ * @param type                       the type of task scheduler, if null then {@link TaskSchedulerType#SEQUENTIAL} is
+ *                                   used
+ * @param unhandledTaskCapacity      the maximum number of unhandled tasks, or 0 if unbounded, if null then 0 is used
```

Comment:
Maybe `1` be a safer default `unhandledTaskCapacity`? A queue that accidentally has a size of 1 when more is needed will simply have bad performance. On the other hand, a queue that is accidentally unbounded could behave fine in the normal case, but be dangerous in the edge case.

Reply:
Alternatively, we could force definition of unhandled task capacity, and not permit any scheduler to rely on a default

Reply:
why not use a primitive for this?

Reply:
I'm ok using 1 as the default for now. In the medium to long term, I expect we are going to make all of them unbounded -- relying on some sort of health monitor to detect when we need to stop gossiping. At that point in time the default will be changed back to unlimited.

Reply:
> why not use a primitive for this?

I'm using a boxed primitive intentionally. This allows `null` to represent when a value was not specified (i.e. when configured to use the default).

Reply:
Change executed. Looking at existing scheduler configuration, the only component utilizing the previous default was the post hash collector. I've now explicitly configured this to use an unlimited capacity.